bundles/org.openhab.binding.freeboxos/src/main/java/org/openhab/binding/freeboxos/internal/handler/FreeDeviceHandler.java

```diff
@@ -0,0 +1,145 @@
+/**
+ * Copyright (c) 2010-2021 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.freeboxos.internal.handler;
+
+import static org.openhab.binding.freeboxos.internal.FreeboxOsBindingConstants.*;
+
+import java.time.ZoneId;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.binding.freeboxos.internal.api.FreeboxException;
+import org.openhab.binding.freeboxos.internal.api.system.DeviceConfig;
+import org.openhab.binding.freeboxos.internal.api.system.Sensor;
+import org.openhab.binding.freeboxos.internal.api.system.Sensor.SensorKind;
+import org.openhab.core.library.CoreItemFactory;
+import org.openhab.core.library.unit.SIUnits;
+import org.openhab.core.library.unit.Units;
+import org.openhab.core.thing.Channel;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.builder.ChannelBuilder;
+import org.openhab.core.thing.type.ChannelTypeUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ *
+ * @author Gaël L'hopital - Initial contribution
+ */
+@NonNullByDefault
+public abstract class FreeDeviceHandler extends HostHandler {
+    private final Logger logger = LoggerFactory.getLogger(FreeDeviceHandler.class);
+    private long uptime = -1;
+
+    public FreeDeviceHandler(Thing thing, ZoneId zoneId) {
+        super(thing, zoneId);
+    }
+
+    @Override
+    public void initialize() {
+        super.initialize();
+        try {
+            while (!checkBridgeHandler()) {
+                Thread.sleep(1000);
+            }
```

Comment:
This is of course a very bad implementation of bridge/child initialization.
I will create for you a proper implementation later.

Reply:
Yes, I kind of agree :) you're more than welcome to help me enhance this.

Reply:
PR submitted to your repo

Reply:
Thanks, I took your PR in account. Did not merged it because I progressed on my side but did integrate all your modifications.